### PR TITLE
[FIX] [8.0]sale_commision: Create only one settlement for period and agent

### PR DESCRIPTION
**Pull request by Daniel-CA**
Fixed bug that creates multiple settlements for same agent on same period.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Strange bug... This seems it was introduced later than the initial commit. Please investigate who was the person and ping here for having their version.

**Comment by Daniel-CA** [comment]
I have reviewed v9.0 to check id the error was there too and looks like that the bug come from the commit of multi-company improvement in v8.0
https://github.com/OCA/commission/commit/9af7f5e0c7e82c4e58770373dc90b1a8abae6278#diff-81a891bc657c17294a6919bb7450d550R69

**Comment by Daniel-CA** [comment]
@jesusVMayor

**Comment by Daniel-CA** [comment]
When this one is merged this one can be closed too #109.

**Comment by Daniel-CA** [comment]
@pedrobaeza, @gurneyalex  I have been watching runbot logs and there is a warning line:
2017-07-20 07:59:50,959 152 WARNING openerp_test openerp.models: method res.partner._check_employee: @constrains parameter 'employee' is not writeable
Could be this the reason to get  :x: in runbot?

**Comment by pedrobaeza** [comment]
Yeah, that's it. hr_commission module should be fixed for avoiding the warning.

File: sale_commission/wizard/wizard_settle.py [hunk 1]
```diff
@@ -69,41 +69,38 @@ def action_settle(self):
...
-                                     'company_id': company.id})
-                                settlement_ids.append(settlement.id)
-                            settlement_line_obj.create(
-                                {'settlement': settlement.id,
-                                 'agent_line': [(6, 0,
-                                                 [agent_lines_company[pos].id])
-                                                ]})
+                agent_lines_company = agent_lines.filtered(
+                    lambda r: r.invoice_line.company_id == company)
+                if agent_lines_company:
```
Comment (pedrobaeza) [inline sale_commission/wizard/wizard_settle.py]: Reduce cyclomatic complexity (and thus, an indentation level), putting:
```
if not agent_lines_company:
    continue
```

**Comment by Daniel-CA** [comment]
Can this one be merged? The coveralls decreased because there is redundant code that could be omitted, but it would give red again. :dizzy_face:

**Comment by pedrobaeza** [comment]
I need someone that checks that the behavior is correct. I'm no more in v8 for checking.

**Review by jesusVMayor — APPROVED** [review]
Tested on runbot. 
The removed loop don't have sense.

**Comment by pedrobaeza** [comment]
One last change: please up module version number

**Comment by Daniel-CA** [comment]
Thanks @jesusVMayor for checking it, change done.

**Pull request merged**